pgmpy/estimators/TreeAugmentedNaiveBayesSearch.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+
+import networkx as nx
+
+from pgmpy.estimators import StructureEstimator
+from sklearn.metrics import mutual_info_score
+from pgmpy.base import DAG
+
+
+class TreeAugmentedNaiveBayesSearch(StructureEstimator):
+    def __init__(self, data, class_node, root_node=None, **kwargs):
+        """
+        Search class for learning tree-augmented naive bayes (TAN) graph structure with a given set of variables.
+
+        Parameters
+        ----------
+        data: pandas DataFrame object
+            dataframe object where each column represents one variable.
+
+        class_node: node
+            Dependent variable of the model (i.e. the class label to predict)
+
+        root_node: node (optional)
+            The root node of the tree structure over the independent variables.  If not specified, then
+            an arbitrary independent variable is selected as the root.
+
+        state_names: dict (optional)
+            A dict indicating, for each variable, the discrete set of states (or values)
+            that the variable can take. If unspecified, the observed values in the data set
+            are taken to be the only possible states.
+
+        complete_samples_only: bool (optional, default `True`)
+            Specifies how to deal with missing data, if present. If set to `True` all rows
+            that contain `np.Nan` somewhere are ignored. If `False` then, for each variable,
+            every row where neither the variable nor its parents are `np.NaN` is used.
+            This sets the behavior of the `state_count`-method.
+
+        References
+        ----------
+        Friedman N, Geiger D and Goldszmidt M (1997). Bayesian network classifiers. Machine Learning 29: 131–163
+
+        """
```

Comment:
I’d be interested in adding documentation to `examples` but I may be a little bit short on time in the coming weeks.  In the interest of not blocking this PR, perhaps I can create a separate ticket to keep track of that task?

Reply:
Since this is the only way for a user to understand how to use the class, I would suggest adding some basic examples before merging this. The same examples that you are adding in the tests would work too. Or else, I can also add the examples if you don't have time for it.